### DISCUSSION TITLE
Disabling Sunshine auto-opening a browser on every boot

### DISCUSSION THREAD

**Discussion by Nitrousoxide** [LizardByte/community#358] [Sunshine] [answered]
Is there a way to stop sunshine from trying to open the config page in the browser on the first time it's started up on a boot?  I don't feel like this is really necessary most of the time.  I'd be perfectly fine having it default to that (since someone who just newly installed it probably needs that) but letting them flip it to "don't open the browser config on startup" when they are done getting it situated seems like a good option to have.

**Comment by ReenigneArcher** [comment]
Sunshine runs in the background as a service. If you click the icon from the start menu it's only opening the browser page.

**Reply by Hixbot2** [reply]
Starting the .desktop file on linux gnome automatically opens the browser page.   I want to start it in the background on startup, but there seems to be no way to avoid the browser page, this is minimizing my fullscreen dosbox.  As far as I know there is no way to have sunshine .desktop run at startup without the browser page. Running the flatpak from terminal also opens the browser page.   How else can the flatpak service run at startup without the browser page opening?

**Reply by Nitrousoxide** [reply]
You can start it through a systemd .service file.  Though it does not reliably start after the desktop is ready in my experience unless you use a .desktop file rather than a .service file with the various .targets.  

```
[Unit]
Description=Sunshine is a self-hosted game stream host for Moonlight.
StartLimitIntervalSec=500
StartLimitBurst=5
After=graphical-session.target

[Service]
ExecStart=flatpak run dev.lizardbyte.app.Sunshine
ExecStop=flatpak kill dev.lizardbyte.app.Sunshine
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=default.target
```

is what my sunshine.service file looks like, but it will frequently start before the desktop is fully ready despite the 
`After=graphical-session.target`

You can add a "wait" for some arbitrary time before the "flatpak run" portion of ExecStart, and that will (usually) fix the issue if you make it long enough. But since it's not really tied to whether the desktop is actually ready yet or not it's a hack and prone to a race condition and can still fail if the desktop is delayed for some reason.

Using a .desktop file does seem to fix the race condition as, at least for gnome, it will **never** try to start the autostart desktop files before the graphical session is ready enough for sunshine.  Though I guess doing it this way brings up the browser window to manage the app.

This is why I wanted a toggle of whether starting the app brings up the browser to config it or not.

**Reply by Hixbot2** [reply]
I have limited success with the flatpak as a systemd service.  I think you and I are in the same boat.  As you say, perhaps a config toggle to choose if you want the web page to load on first run.   Or perhaps a command line argument to run the .desktop without the browser starting.

**Reply by ReenigneArcher** [reply]
I originally assumed you were running Sunshine from Windows. Apologies.

Easiest way to get the behavior changed is to open a PR, and we will review.

**Reply by mattskiiau** [reply]
Also having this issue on steam deck. Pops up every boot or swap to desktop mode from gaming mode.

**Comment by DraakjeYoblama** [comment]
> Sunshine runs in the background as a service. If you click the icon from the start menu it's only opening the browser page.

Does this mean there is a way to start sunshine on Windows without the browser opening? Or am I misinterpreting that?

Edit: I found it already, turns out I wasn't running autostart-service.bat as an administrator. But I didn't see the error because of the @echo off.

**Reply by ReenigneArcher** [reply]
Better to just use the installer on Windows.

**Comment by davidcm** [comment]
Hi, Windows user here.

I installed Sunshine via installer, disabling the option to start automatically. My most convenient workflow is to start Sunshine from the start menu, then a script in my Raspberry Pi detects the server and automatically runs Moonlight.

Everything works, but it's how Sunshine starts that doesn't feel quite elegant. First it shows the console, then Windows pops up an alert asking for permission to run sunshine.exe, then another console opens, then it automatically opens the web UI in Firefox.

Those are a lot of things that a normal user doesn't really need, when I'd only wish to click the Sunshine icon and have the tray icon appear.

Do you think if by any chance there's a possibility to smooth out the launching process?

Thanks!

**Reply by cgutman** [reply]
> Everything works, but it's how Sunshine starts that doesn't feel quite elegant. First it shows the console, then Windows pops up an alert asking for permission to run sunshine.exe, then another console opens, then it automatically opens the web UI in Firefox.

We could potentially suppress one of those console windows or add separate shortcuts for the Web UI and background launch options, but no the launch experience can't really be much simpler that that. We must elevate to admin to start the service, so there's no way around that UAC prompt unless you want to disable UAC entirely on your system.

You could run the portable build without elevation, but then you won't be able to interact with any admin apps at all and performance will be degraded due to lack of permissions for Sunshine to increase its GPU priority.

**Reply by ld892012** [reply]
You can create a scheduled task to run manually Sunshine service without UAC and it will not open browser ^^

**Reply by Dalai-** [reply]
Was there ever a fix for this? It's super annoying to have the web UI popup every time I manually start Sunshine.

**Reply by terrapass** [reply]
@Dalai- Not sure about a fix, but a partial workaround I found on Windows 11 is to modify the Sunshine start menu shortcut to start the Sunshine service, instead of running `sunshine.exe --shortcut`, as it does by default.

Here are the steps I used:
1. Run `install-service.bat` as administrator (in my case it was located in `C:\Program Files\Sunshine\scripts`).
2. Find the shortcut you're using to run Sunshine (in my case inside `C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Sunshine`) and open its Properties (or make a copy and change its Properties, if you prefer).
3. Set Target to the following:
```
C:\Windows\System32\net.exe start "Sunshine Service"
```
4. Go to "Advanced..." and check "Run as administrator".
5. (optional) Restore the Sunshine icon by using "Change Icon..." and browsing to `sunshine.exe`, so the shortcut continues to look nice.

Now when I run Sunshine via this shortcut, a browser window with Web UI doesn't pop up. A downside is that UAC confirmation is displayed when you run the shortcut, and a console window can be seen for a couple of seconds as the service starts.

But at least it doesn't reopen Chrome and my 100+ tabs therein to needlessly display the Web UI now. 😅

**Comment by ld892012** [comment]
If you want manually start Sunshine without open a browser and without UAC prompt you can create a scheduled task that can be manually started with a shortcut :
- Go to Task Scheduler
- Click on "Create Task"
- Set a name ("Sunshine" for example)
- Check "Run with highest privileges"
- Go to  "Actions" tab
- Click on "New" button
- In "Program/script:" paste `C:\Windows\System32\net.exe`
- In "Add arguments (optional):" paste `start SunshineService`
- Click on OK button
- Go to "Settings" tab and ensure "Allow the task to run on demand" is checked
- Click on OK button

Now your scheduled task is ready, you can now start Sunshine with cmd with this command :
`schtasks.exe /RUN /TN <name of scheduled task>`

If you want to create a shortcut :
- Right click on desktop or in a folder
- Click on "New" option then select "Shortcut"
- In "Type the location of the item" paste `C:\Windows\System32\schtasks.exe /RUN /TN <name of scheduled task>`
- Go to Next
- Type the name you want then click Finish

You can move this shortcut to the Start Menu :
- Open Explorer
- Click on location bar, paste `%AppData%\Microsoft\Windows\Start Menu\Programs` then press Enter
- Just move your shortcut in this folder

You can assign an icon for your shortcut with `Properties/Shortcut/Change Icon`

**Reply by GuyWhoDidThatThing** [reply]
This solution has one small issue for me. It does exactly as expected in that Sunshine is open on login automatically with no _visible_ browser window. However, my Firefox then won't open due to an already running invisible process. When I force close this and re-open Firefox I'm forced to the Sunshine configuration page. Not sure if this happens with other browsers, but it is annoying to have to end that process on every boot. Is there no arg to stop Sunshine from auto-opening the configuration page in general?

**Reply by darmach** [reply]
Come on guys... it's 2026. Just add a switch to suppress the web ui opening. The way this starts is really uncivilized.

**Reply by ReenigneArcher** [reply]
Happy New Year

We accept PRs.

**Reply by darmach** [reply]
@ReenigneArcher Haha, classic! >_< Touché!

Sarcasm aside, you are probably aware that is will be more complicated from the user perspective - me not being familiar with the codebase - than someone already working on the project. Would it be feasible to implement a command line parameter along `--shortcut` on Windows to not show web UI on app start? Or too much legacy code around to dig - I would imagine this can make seemingly simple features to me easily added, it often makes them hard in projects I'm working in.

**Reply by ashvinau** [reply]
I use sunshine on a media center PC which i control with a game controller from 10 feet away. After this most recent update, i was baffled to see the config page start on every boot. It never did that before and I searched exhaustively for a way to disable the browser pop-up. I was pretty dismayed to find this thread and discover that not only is there no way to do this, but the contributors actively refuse to even consider such a thing for whatever reason. 

The media center has no conventional mouse and keyboard. I use steam big picture and shortcuts for Kodi and Retroarch. I have no way of closing the browser window on boot but to plug a mouse in and close the browser. 

This renders sunshine unusable for me, and I will have to uninstall even though I like the functionality the software brings.

I am a programmer, and I will take a look at the code and see if i can fix it i guess. I would imagine I can search the codebase for the literal strings in the UI, and work backwards to see where it is used. Doesn't seem to be any html templates, so I would imagine it's all inline. 

If i put in a pull request will this team actually accept it? Or is this not worth my time?

**Reply by ReenigneArcher** [reply]
> It never did that before

This has happened for years for Flatpak and the Windows Service. It doesn't happen for any other versions.

**Comment by ReenigneArcher** [comment, marked as answer]
Should be solved by https://github.com/LizardByte/Sunshine/pull/5329

**Comment by lordgeyik** [comment]
This worked for me on Linux Mint:
- Create a startup item (choose Sunshine application)
- Use this command: /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=/app/bin/sunshine dev.lizardbyte.app.Sunshine

This will start Sunshine in the background and NOT open the web app in the browser automatically.

<img width="1816" height="478" alt="image" src="https://github.com/user-attachments/assets/b89a7451-1e8a-4c2e-94a8-1ad2efb2725e" />
